src/static_array.cr

```diff
@@ -25,6 +25,17 @@ struct StaticArray(T, N)
     new { value }
   end
 
+  def ==(other : self)
+    length == other.length && each_with_index do |e, i|
+      return false unless other[i] == e
```

Comment:
Minor detail but I'd probably do `e == other[i]`. Given `==` is a method call I think it's more expected for the elements of the receiver to receive that `==` method call, think you have your custom objects in it with a `==` implementation that's more permissive than that of the ones you're comparing to.